### PR TITLE
produce an error when Quarto content cannot be inspected

### LINKED ISSUE #1032: quarto: provide feedback when inspect fails
Given the Quarto document:

````markdown
---
format: unsupported
---

This is a Quarto document that attempts to use an unrecognized format.
````

Running `quarto inspect` produces an error.

```bash
quarto inspect index.qmd
# => ERROR: Unknown format unsupported
```

However, when `rsconnect::writeManifest()` or `rsconnect:deployApp()` acts upon this file, there is no complaint.

```r
rsconnect::writeManifest()
```

The manifest declares that it should be `quarto-static` but is created without a `quarto` block. Additionally, a set of R packages is inferred, as if it needs knitr/rmarkdown.

When `rsconnect::deployApp()` publishes to Posit Connect, the lack of a Quarto block produces an error:

```
── Deployment complete ───
✖ Deployment failed with error: Unsupported App Mode.
```

This type of error can happen when the document uses a new type of format but the Quarto installation does not support that format.

### PR DESCRIPTION
**Pull request by aronatkins**
fixes #1032

Note: Builds upon https://github.com/rstudio/rsconnect/pull/1038.

### CODE REVIEW

File: R/appMetadata-quarto.R [hunk 1]
```diff
@@ -37,20 +34,34 @@ quartoInspect <- function(appDir = NULL, appPrimaryDoc = NULL) {
...
+    cli::cli_abort(
+      c(
+        "Unable to run `quarto inspect` against your content:",
+        json
+      )
     )
   }
-  return(NULL)
+  parsed <- jsonlite::fromJSON(json)
+  return(parsed)
```
Comment (hadley) [inline R/appMetadata-quarto.R:64]:
```suggestion
  jsonlite::fromJSON(json)
```

File: tests/testthat/test-appMetadata-quarto.R [hunk 2]
```diff
@@ -123,9 +99,42 @@ test_that("quartoInspect processes content with filenames containing spaces", {
...
+test_that("quartoInspect produces an error when a document cannot be inspected", {
+  skip_if_no_quarto()
+
+  dir <- local_temp_app(list("bad.qmd" = c(
+    "---",
+    "format: unsupported",
+    "---",
+    "this is a document using an unsupported format."
+  )))
+  expect_error(
```
Comment (hadley) [inline tests/testthat/test-appMetadata-quarto.R:111]: I have a very mild preference for `expect_snapshot()` here so it's a bit easier to update if the error ever changes and so you can see if the correct call is bubbled up in the error.

File: R/appMetadata-quarto.R [hunk 3]
```diff
@@ -37,20 +34,34 @@ quartoInspect <- function(appDir = NULL, appPrimaryDoc = NULL) {
...
-      {
-        json <- suppressWarnings(system2(quarto, args, stdout = TRUE, stderr = TRUE))
-        parsed <- jsonlite::fromJSON(json)
-        return(parsed)
-      },
-      error = function(e) NULL
+  if (!is.null(status)) {
+    cli::cli_abort(
+      c(
+        "Unable to run `quarto inspect` against your content:",
```
Comment (hadley) [inline R/appMetadata-quarto.R:58]:
```suggestion
        "Failed to run `quarto inspect` against your content:",
```

I don't know why but my heart wants "failed" here

**Review by hadley — APPROVED** [review]

File: tests/testthat/test-appMetadata-quarto.R (hunk 2 above)
Comment (aronatkins) [reply tests/testthat/test-appMetadata-quarto.R:111]: The challenge here is that Quarto emits a stack trace with source references. I'll use `expect_snapshot()` and throw a `transform` function into the mix.

File: R/appMetadata-quarto.R (hunk 1 above)
Comment (aronatkins) [reply R/appMetadata-quarto.R]: Done.

File: tests/testthat/test-appMetadata-quarto.R (hunk 2 above)
Comment (aronatkins) [reply tests/testthat/test-appMetadata-quarto.R]: Done. Blindly `head(lines,3)` and using that as the important information.

File: R/appMetadata-quarto.R (hunk 3 above)
Comment (aronatkins) [reply R/appMetadata-quarto.R]: I'm OK with making your ❤️ happy. Done.

File: tests/testthat/test-appMetadata-quarto.R (hunk 2 above)
Comment (aronatkins) [reply tests/testthat/test-appMetadata-quarto.R]: Also: `NOCOLOR=1` to suppress colors from Quarto error output.

**Review by toph-allen — COMMENTED** [review]
Thanks for tackling this when I was busy. 🙇🏻‍♂️

**Pull request merged**